kdt-board/src/main/java/com/example/kdtboard/service/PostService.java

```diff
@@ -0,0 +1,58 @@
+package com.example.kdtboard.service;
+
+import com.example.kdtboard.converter.PostConverter;
+import com.example.kdtboard.domain.Post;
+import com.example.kdtboard.repository.PostRepository;
+import com.example.kdtboard.domain.User;
+import com.example.kdtboard.repository.UserRepository;
+import com.example.kdtboard.dto.CreatePostRequest;
+import com.example.kdtboard.dto.PostDto;
+import com.example.kdtboard.dto.UpdatePostRequest;
+import javassist.NotFoundException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+
+@Service
+public class PostService {
+    @Autowired
+    private PostRepository postRepository;
+    @Autowired
+    private UserRepository userRepository;
+
+    @Autowired
+    private PostConverter postConverter;
+
+    @Transactional
+    public Long save(CreatePostRequest postRequest) throws NotFoundException {
+        User user = userRepository.findById(postRequest.getUserId())
+                .orElseThrow(() -> new NotFoundException("Invalid User, UserId: " + postRequest.getUserId()));
+        Post post = postConverter.convertPostWithUser(postRequest,user);
+        Post savedPost = postRepository.save(post);
+        return savedPost.getId();
+    }
+
+    @Transactional
+    public Page<PostDto> findAll(Pageable pageable){
+        return postRepository.findAll(pageable)
+                .map(postConverter::convertPostDto);
+    }
+
+    @Transactional
+    public PostDto findOne(Long id) throws NotFoundException {
```

Comment:
```suggestion
    @Transactional(readOnly = true)
    public Page<PostDto> findAll(Pageable pageable){
        return postRepository.findAll(pageable)
                .map(postConverter::convertPostDto);
    }

    @Transactional(readOnly = true)
    public PostDto findOne(Long id) throws NotFoundException {
```
조회용으로 보이는데요. 방어적으로 readOnly를 걸어놓으면, 다른사람이 코드를 수정했을 때 데이터의 변경을 방어할 수 있을 것 같아요.